### PR TITLE
Medianet: Adds mtype support to be used as BidType

### PR DESCRIPTION
**Pull request by amanpatniajmer**

### CODE REVIEW

File: adapters/medianet/medianet.go [hunk 1]
```diff
@@ -89,8 +89,13 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server co
...
 }
 
-func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
-	mediaType := openrtb_ext.BidTypeBanner
+func getMediaTypeForImp(bid *openrtb2.Bid, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	mediaType, err := getBidMediaTypeFromMtype(bid)
+	if err == nil {
+		return mediaType, nil
+	}
+	mediaType = openrtb_ext.BidTypeBanner
```
Comment (gargcreation1992) [inline adapters/medianet/medianet.go:97]: Should remove this check. If Mtype is not set then please return with err.

File: adapters/medianet/medianet.go [hunk 2]
```diff
@@ -89,8 +89,13 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server co
 	}, nil
 }
 
-func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
-	mediaType := openrtb_ext.BidTypeBanner
+func getMediaTypeForImp(bid *openrtb2.Bid, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	mediaType, err := getBidMediaTypeFromMtype(bid)
+	if err == nil {
+		return mediaType, nil
```
Comment (gargcreation1992) [inline adapters/medianet/medianet.go:95]: Should not ignore error returned from `getBidMediaTypeFromMtype` function. This should be - 
`if err != nil {
     return nil, err
}`

File: adapters/medianet/medianet.go (hunk 1 above)
Comment (amanpatniajmer) [reply adapters/medianet/medianet.go]: Removed the entire fallback. Pls check now

File: adapters/medianet/medianet.go (hunk 2 above)
Comment (amanpatniajmer) [reply adapters/medianet/medianet.go]: Removed the fallback. Pls check

**Review by Sonali-More-Xandr — APPROVED** [review]

**Review by gargcreation1992 — APPROVED** [review]

**Pull request merged**